### PR TITLE
feat(controller update): Return data with associated data if an include is defined

### PR DESCRIPTION
**Pull request by Fridus**
- Update controller `update`
- add tests

### CODE REVIEW

File: tests/resource/associations.test.js [hunk 1]
```diff
@@ -99,6 +99,12 @@ describe('Resource(associations)', function() {
           endpoints: ['/addresses', '/addresses/:id']
         });
 
+        rest.resource({
```
Comment (mbroadst) [inline tests/resource/associations.test.js]: this resource doesn't look like its used anywhere

Comment (Fridus) [inline tests/resource/associations.test.js]: Oh sorry !

**Comment by mbroadst** [comment]
this looks really good to me, the only thing I would say is that it changes the output people may have been unwittingly depending upon up until now. I am planning to make a v1.0.0 release with all this new code in master, however since we're already going to be including the auto-associations branch maybe now is a good time to introduce the "associations" option to the resource factory. This way you could toggle this behavior on or off, and default to off to maintain existing code. Thoughts?

**Comment by Fridus** [comment]
For v1.0.0, it's good :)
I think it is not a good idea to make a toggle. The problem for retro-compatibility is the key of associate object is removed by https://github.com/dchester/epilogue/blob/master/lib/Controllers/read.js#L33-L39 .
Maybe we can change it for keep this reference ?

**Comment by Fridus** [comment]
If I remove `includeAttributes` for `base`, `read` and `list`, this is maybe the solution

**Comment by mbroadst** [comment]
IMHO we absolutely have to make the behavior toggle-able: not everyone wants to return associated data during reads (even if they specify an include). Maybe they just want to add include data for list operations?  It's also not very difficult to add this in, because "associations" in the resource definition are going to have to grow to include a number of cases, e.g. "what operations do you want to auto-associate for", or "which includes do you want to auto-associate for", "should associations be followed at all", etc

**Comment by Fridus** [comment]
Here I'm just reapply the default return of `Read`.
The only one difference is, in case of inclusion, it return the associated data instead of the identifier of association. So, if we define a resource with associations, it is normal to return the associated data.
If we dont't want this data, we can define another resource without the association.
Anyway, I think it's smarter to keep the identifier to addition of associated data. It makes the information compatible with old versions.

Ex:
```
{
  "id": 4,
  "title": "Task 4",
  "description": "... 4",
  "deadline": null,
  "createdAt": "2015-06-02T16:00:05.000Z",
  "updatedAt": "2015-06-02T16:03:16.000Z",
  "projectId": 1, // identifier
  "project": {    // data
    "id": 1,
    "title": "Project 1",
    "description": "description 1",
    "createdAt": "2015-06-02 16:00:05.000 +00:00",
    "updatedAt": "2015-06-02 16:00:05.000 +00:00"
  }
}
```
To sum up:
- if the resource has not the association, the result is the same as before.
- if the resource has the association, the associated data are included and and this is, for me, more logical.

**Comment by mbroadst** [comment]
Yes I understand what the code does, but from a legacy support perspective you are changing the output of this operation and legacy users may depend on that output. My proposed solution is to provide options under options.associations that would allow you to toggle any or all of this behavior:

 * It was by design that the identifiers were dropped when associated data was included, this was to satisfy my use case at the time with the added (marginal) benefit of reducing the payload returned by the server. I personally prefer this style, but this very well could be toggled by e.g. ```options.associations.dropIdentifier``` or something

 * I don't consider just adding the identifiers back in as a proper solution to this because there might be cases where a user wants to include associated data in e.g. the ListController, but not for the ReadController (and we shouldn't force them to declare a new resource just for this case). It's not too difficult to add a toggle in the association options to conditionally include related data in any given controller.

TL;DR: I understand the idea that adding a include statement in your resource definition would seem to imply that you always want to include the associated data, but I don't think we can make this assumption and it's easy enough to make this configurable (perhaps with a sensible default of automatically including the related data).
